python/lsst/analysis/drp/scatterPlot.py

```diff
@@ -0,0 +1,532 @@
+        in each patch for the column in self.zColName.
+
+        `scatterPlotWithTwoHists` which makes a scatter plot of the points with
+        a histogram of each axis.
+        """
+
+        # Apply the selectors to narrow down the sources to use
+        mask = np.ones(len(catPlot), dtype=bool)
+        for selector in self.config.selectorActions:
+            mask &= selector(catPlot)
+        catPlot = catPlot[mask]
+
+        columns = {self.config.axisLabels["x"]: self.config.axisActions.xAction(catPlot),
+                   self.config.axisLabels["y"]: self.config.axisActions.yAction(catPlot),
+                   self.config.axisLabels["mag"]: self.config.axisActions.magAction(catPlot),
+                   "patchId": catPlot["patchId"]}
+        for actionStruct in [self.config.highSnStatisticSelectorActions,
+                             self.config.lowSnStatisticSelectorActions,
+                             self.config.sourceSelectorActions]:
+            for action in actionStruct:
+                for col in action.columns:
+                    columns.update({col: catPlot[col]})
+
+        plotDf = pd.DataFrame(columns)
+
+        sourceTypes = np.zeros(len(plotDf))
+        for selector in self.config.sourceSelectorActions:
+            # The source selectors return 1 for a star and 2 for a galaxy
+            # rather than a mask this allows the information about which
+            # type of sources are being plotted to be propagated
+            sourceTypes += selector(catPlot)
+        plotDf.loc[:, "sourceType"] = sourceTypes
+
+        # Decide which points to use for stats calculation
+        useForStats = np.zeros(len(plotDf))
+        lowSnMask = np.ones(len(plotDf), dtype=bool)
+        for selector in self.config.lowSnStatisticSelectorActions:
+            lowSnMask &= selector(plotDf)
+        useForStats[lowSnMask] = 2
+
+        highSnMask = np.ones(len(plotDf), dtype=bool)
+        for selector in self.config.highSnStatisticSelectorActions:
+            highSnMask &= selector(plotDf)
+        useForStats[highSnMask] = 1
+        plotDf.loc[:, "useForStats"] = useForStats
+
+        # Get useful information about the plot
+        plotInfo = parsePlotInfo(dataId, runName, tableName)
+        # Calculate the corners of the patches and some associated stats
+        sumStats = generateSummaryStats(plotDf, self.config.axisLabels["y"], skymap, plotInfo)
+        # Make the plot
+        fig = self.scatterPlotWithTwoHists(plotDf, plotInfo, sumStats)
+
+        return pipeBase.Struct(scatterPlot=fig)
+
+    def scatterPlotWithTwoHists(self, catPlot, plotInfo, sumStats, yLims=False, xLims=False):
+
+        """Makes a generic plot with a 2D histogram and collapsed histograms of
+        each axis.
+
+        Parameters
+        ----------
+        catPlot : `pandas.core.frame.DataFrame`
+            The catalog to plot the points from.
+        xLabel : `str`
```

Comment:
update docstring to reflect actual parameters. `xLabel` and `yLabel` are not here.